### PR TITLE
Fix sonar warnings

### PR DESCRIPTION
**Pull request by osmandin**
https://jira.duraspace.org/browse/FCREPO-1320

### CODE REVIEW

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/observer/eventmappings/AllNodeEventsOneEvent.java [hunk 1]
```diff
@@ -70,59 +70,68 @@ public String apply(final Event ev) {
...
 
-        return new Iterator<FedoraEvent>() {
+    private class FedoraEventIterator implements Iterator {
 
-            // sort JCR events into a Multimap keyed by the node ID involved
-            final Multimap<String, Event> sortedEvents = index(events, EXTRACT_NODE_ID);
+        final Iterator<Event> events;
 
-            final Iterator<String> nodeIds = sortedEvents.keySet().iterator();
+        public FedoraEventIterator(final Iterator<Event> events) {
```
Comment (awoods) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/observer/eventmappings/AllNodeEventsOneEvent.java]: Please move the constructor below the following two instance variables:
`final Multimap<String, Event> sortedEvents;
 final Iterator<String> nodeIds;`

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/ChildrenRdfContext.java [hunk 2]
```diff
@@ -72,7 +72,6 @@ public ChildrenRdfContext(final FedoraResource resource,
 
             @Override
             public Iterator<Triple> apply(final FedoraResource child) {
-
```
Comment (awoods) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/ChildrenRdfContext.java]: sneaky

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/FixityRdfContext.java [hunk 3]
```diff
@@ -71,48 +71,65 @@ public FixityRdfContext(final FedoraResource resource,
         super(resource, idTranslator);
 
         concat(Iterators.concat(Iterators.transform(blobs.iterator(),
-                new Function<FixityResult, Iterator<Triple>>() {
+                new FixityResultIteratorFunction(resource, idTranslator, digest, size))));
+    }
+
+    private class FixityResultIteratorFunction implements Function<FixityResult, Iterator<Triple>> {
+
+        final FedoraResource resource;
```
Comment (awoods) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/FixityRdfContext.java]: These should all be `private`, no?

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/LdpContainerRdfContext.java [hunk 4]
```diff
@@ -147,47 +147,62 @@ public boolean apply(final Property property) {
...
-                        childSubject = translator().reverse()
-                                .convert(((NonRdfSourceDescription) child).getDescribedResource())
-                                .asNode();
-                    } else {
-                        childSubject = translator().reverse().convert(child).asNode();
-                    }
+    private class FedoraResourceTripleFunction implements Function<FedoraResource, Iterator<Triple>>   {
 
-                    if (insertedContainerProperty.equals(MEMBER_SUBJECT.getURI())) {
+        final String insertedContainerProperty;
```
Comment (awoods) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/LdpContainerRdfContext.java]: `private`

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/LdpContainerRdfContext.java [hunk 5]
```diff
@@ -147,47 +147,62 @@ public boolean apply(final Property property) {
...
 
-                        return singletonIterator(create(subject(), memberRelation, childSubject));
-                    }
-                    final String insertedContentProperty = getPropertyNameFromPredicate(resource().getNode(),
-                            createResource(insertedContainerProperty),
-                            null);
+        final com.hp.hpl.jena.graph.Node memberRelation;
```
Comment (awoods) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/LdpContainerRdfContext.java]: `private`

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/observer/eventmappings/AllNodeEventsOneEvent.java [hunk 6]
```diff
@@ -70,59 +70,68 @@ public String apply(final Event ev) {
...
+            sortedEvents = index(events, EXTRACT_NODE_ID);
+            nodeIds = sortedEvents.keySet().iterator();
+        }
 
-            @Override
-            public boolean hasNext() {
-                return nodeIds.hasNext();
-            }
+        // sort JCR events into a Multimap keyed by the node ID involved
+        final Multimap<String, Event> sortedEvents;
```
Comment (awoods) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/observer/eventmappings/AllNodeEventsOneEvent.java]: `private`

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/observer/eventmappings/AllNodeEventsOneEvent.java [hunk 7]
```diff
@@ -70,59 +70,68 @@ public String apply(final Event ev) {
...
     public Iterator<FedoraEvent> apply(final Iterator<Event> events) {
+        return new FedoraEventIterator(events);
+    }
 
-        return new Iterator<FedoraEvent>() {
+    private class FedoraEventIterator implements Iterator {
 
-            // sort JCR events into a Multimap keyed by the node ID involved
-            final Multimap<String, Event> sortedEvents = index(events, EXTRACT_NODE_ID);
+        final Iterator<Event> events;
```
Comment (awoods) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/observer/eventmappings/AllNodeEventsOneEvent.java]: `private`

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/observer/eventmappings/AllNodeEventsOneEvent.java [hunk 8]
```diff
@@ -70,59 +70,68 @@ public String apply(final Event ev) {
...
-                while (nodeSpecificEvents.hasNext()) {
-                    // add the event type and property name to the event we are building up to emit
-                    // we could aggregate other information here if that seems useful
-                    final Event otherEvent = nodeSpecificEvents.next();
-                    fedoraEvent.addType(otherEvent.getType());
-                    addProperty(fedoraEvent, otherEvent);
-                }
-                return fedoraEvent;
-            }
+        final Iterator<String> nodeIds;
```
Comment (awoods) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/observer/eventmappings/AllNodeEventsOneEvent.java]: `private`

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/FixityRdfContext.java [hunk 9]
```diff
@@ -71,48 +71,65 @@ public FixityRdfContext(final FedoraResource resource,
...
-                                    contentLocation));
-                            b.add(create(contentLocation,
-                                    RDF.type.asNode(),
-                                    CONTENT_LOCATION_TYPE.asNode()));
-                            b.add(create(contentLocation,
-                                    HAS_CONTENT_LOCATION_VALUE.asNode(),
-                                    createLiteral(storeIdentifier)));
+                return b.build().iterator();
+            } catch (final RepositoryException e) {
+                throw propagate(e);
```
Comment (ajs6f) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/FixityRdfContext.java]: Throw a `RepositoryRuntimeException` here.

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/utils/FedoraTypesUtils.java [hunk 10]
```diff
@@ -129,18 +129,11 @@ public boolean apply(final Property p) {
             @Override
             public boolean apply(final Property p) {
                 try {
-                    if (!p.getDefinition().isProtected()) {
```
Comment (awoods) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/utils/FedoraTypesUtils.java]: I would suggest leaving the comment in the codebase, even if that means making this Predicate a private class.

**Comment by awoods** [comment]
Resolved with: https://github.com/fcrepo4/fcrepo4/commit/d9a596b8c12bd473038e9455ab83f5d8537f0194

**Pull request closed without merging**